File: Doc/支付接口/即时到账开发文档下载/简易支付插件/java/java/src/com/tenpay/PayResponseHandler.java
package com.tenpay;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.tenpay.util.MD5Util;
import com.tenpay.util.TenpayUtil;

/**
 * 即时到帐应答类
 * ============================================================================
 * api说明：
 * getKey()/setKey(),获取/设置密钥
 * getParameter()/setParameter(),获取/设置参数值
 * getAllParameters(),获取所有参数
 * isTenpaySign(),是否财付通签名,true:是 false:否
 * doShow(),显示处理结果
 * getDebugInfo(),获取debug信息
 * 
 * ============================================================================
 *
 */
public class PayResponseHandler extends ResponseHandler {

	public PayResponseHandler(HttpServletRequest request,
			HttpServletResponse response) {
		
		super(request, response);
		
	}

	/**
	 * 是否财付通签名
	 * @Override
	 * @return boolean
	 */
	public boolean isTenpaySign() {
		
		//获取参数
		String cmdno = this.getParameter("cmdno");
		String pay_result = this.getParameter("pay_result");
		String date = this.getParameter("date");
		String transaction_id = this.getParameter("transaction_id");
		String sp_billno = this.getParameter("sp_billno");
		String total_fee = this.getParameter("total_fee");		
		String fee_type = this.getParameter("fee_type");
		String attach = this.getParameter("attach");
		String key = this.getKey();
		String tenpaySign = this.getParameter("sign").toLowerCase();
		
		//组织签名串
		StringBuffer sb = new StringBuffer();
		sb.append("cmdno=" + cmdno + "&");
		sb.append("pay_result=" + pay_result + "&");
		sb.append("date=" + date + "&");
		sb.append("transaction_id=" + transaction_id + "&");
		sb.append("sp_billno=" + sp_billno + "&");
		sb.append("total_fee=" + total_fee + "&");
		sb.append("fee_type=" + fee_type + "&");
		sb.append("attach=" + attach + "&");
		sb.append("key=" + key);
		
		String enc = TenpayUtil.getCharacterEncoding(
				this.getHttpServletRequest(), this.getHttpServletResponse());
		//算出摘要
		String sign = MD5Util.MD5Encode(sb.toString(), enc).toLowerCase();
		
		//debug信息
		this.setDebugInfo(sb.toString() + " => sign:" + sign +
				" tenpaySign:" + tenpaySign);
		
		return tenpaySign.equals(sign);
	} 
	
}

File: Doc/支付接口/即时到账开发文档下载/简易支付插件/C#/tenpayApp/App_Code/tenpayLib/PayResponseHandler.cs
using System;
using System.Collections;
using System.Text;
using System.Web;
namespace tenpayApp
{
	/// <summary>
	/// PayResponseHandler 的摘要说明。
	/// </summary>
	/**
	* 即时到帐应答类
	* ============================================================================
	* api说明：
	* getKey()/setKey(),获取/设置密钥
	* getParameter()/setParameter(),获取/设置参数值
	* getAllParameters(),获取所有参数
	* isTenpaySign(),是否财付通签名,true:是 false:否
	* doShow(),显示处理结果
	* getDebugInfo(),获取debug信息
	* 
	* ============================================================================
	*
	*/
	public class PayResponseHandler:ResponseHandler
	{
		public PayResponseHandler(HttpContext httpContext) : base(httpContext)
		{
			//
			// TODO: 在此处添加构造函数逻辑
			//
		}
		/**
	 * 是否财付通签名
	 * @Override
	 * @return boolean
	 */
		
		public override Boolean isTenpaySign() 
		{
		
			//获取参数
			string cmdno = getParameter("cmdno");
			string pay_result = getParameter("pay_result");
			string date = getParameter("date");
			string transaction_id = getParameter("transaction_id");
			string sp_billno = getParameter("sp_billno");
			string total_fee = getParameter("total_fee");		
			string fee_type = getParameter("fee_type");
			string attach = getParameter("attach");
			string tenpaySign = getParameter("sign").ToUpper();
			string key = this.getKey();
			
			//组织签名串
			StringBuilder sb = new StringBuilder();
			sb.Append("cmdno=" + cmdno + "&");
			sb.Append("pay_result=" + pay_result + "&");
			sb.Append("date=" + date + "&");
			sb.Append("transaction_id=" + transaction_id + "&");
			sb.Append("sp_billno=" + sp_billno + "&");
			sb.Append("total_fee=" + total_fee + "&");
			sb.Append("fee_type=" + fee_type + "&");
			sb.Append("attach=" + attach + "&");
			sb.Append("key=" + key);
		
			//算出摘要
			string sign = MD5Util.GetMD5(sb.ToString(),getCharset());	

			//debug信息
			setDebugInfo(sb.ToString() + " => sign:" + sign +
				" tenpaySign:" + tenpaySign);
		
			 return sign.Equals(tenpaySign);
		} 
	
	
	}
}
